pkg/api/analyze.go

```diff
@@ -19,4 +19,6 @@ type DetectionResult struct {
 	DevfileVersion   string `json:"devfileVersion,omitempty"`
 	// Name represents the project/application name as detected by alizer
 	Name string `json:"name,omitempty"`
+	// Architectures represent the architectures with which the Devfile must be compatible with.
+	Architectures []string
```

Comment:
This will consequently be exposed in the output of `odo analyze -o json`:
```json
[
  {
    "devfile": "dotnet50",
    "devfileRegistry": "StagingRegistry",
    "devfileVersion": "1.0.3",
    "name": "my-dotnet",
    "Architectures": null
  }
]
```

I think it is okay, but we should consistently lower the field case and omit the field if it is empty.